src/transforms/throttle.rs

```diff
@@ -59,13 +86,21 @@ impl TransformConfig for ThrottleConfig {
     }
 }
 
+#[derive(Clone)]
+pub struct ThrottleUnitBytesEncoding {
+    transformer: Transformer,
+    encoder: Encoder<()>,
+}
+
 #[derive(Clone)]
 pub struct Throttle<C: clock::Clock<Instant = I>, I: clock::Reference> {
     quota: Quota,
     flush_keys_interval: Duration,
     key_field: Option<Template>,
     exclude: Option<Condition>,
     clock: C,
+    unit: ThrottleUnit,
+    unit_bytes_encoding: Option<ThrottleUnitBytesEncoding>,
```

Comment:
Ditto using an `enum` here: The `encoding` bit is needed if and only if the mode is set to bytes, so include it in a new enum for this post-configuration setting:
```rust
enum ThrottleParsedMode {
    Events,
    Bytes(ThrottleUnitBytesEncoding),
}
```
You could even get rid of the `struct ThrottleUnitBytesEncoding` this way by embedding the fields into this enum as above.